File: 33.PictureInPictureDemo/app/src/main/java/com/ani/pictureinpicturedemo/MainActivity.java
package com.ani.pictureinpicturedemo;

import android.content.res.Configuration;
import android.os.Build;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    public void goPiP(View view)
    {
        enterPictureInPictureMode();
    }

    @Override
    public void onPictureInPictureModeChanged(boolean isInPictureInPictureMode, Configuration newConfig) {
        super.onPictureInPictureModeChanged(isInPictureInPictureMode, newConfig);

        TextView textView = findViewById(R.id.textView);
        Button button = findViewById(R.id.button);

        if(isInPictureInPictureMode)
        {
            //Going into PIP

            textView.setText("Hello Ani,\n You are into PiP mode");
            getSupportActionBar().hide();
            button.setVisibility(View.INVISIBLE);
        }
        else
        {
            //Going out of PIP

            textView.setText("Hello Ani");
            getSupportActionBar().show();
            button.setVisibility(View.VISIBLE);
        }
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
}

File: 29.KotlinDemo/app/src/main/java/com/ani/kotlindemo/MainActivity.kt
package com.ani.kotlindemo

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.TextView
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    var counter  = 0


    fun count(view :View){
        counter += 1
        counterTextView.text = counter.toString()
    }

    fun reset(view :View){
        counter = 0
        counterTextView.text = counter.toString()
    }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var counterTextView = findViewById<TextView>(R.id.counterTextView)
    }
}
